storage/fanout.go

```diff
@@ -147,6 +147,12 @@ type fanoutAppender struct {
 	secondaries []Appender
 }
 
+// SetHints is a no-op. It exists to satisfy the Appender interface,
+// but hints are not propagated or stored in this fanoutAppender.
+func (f *fanoutAppender) SetHints(hints *AppendHints) {
+	// Do nothing.
+}
```

Comment:
This is counter-intuitive. I would expect this to set the hints on `primary` and all `secondaries`. Why is it doing nothing?